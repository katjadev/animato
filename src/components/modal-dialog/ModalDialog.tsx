import { FC, ReactNode, useEffect, useRef } from 'react'
import { Inter } from 'next/font/google'
import Icon from '@animato/components/icon/Icon'
import IconButton from '@animato/components/icon-button/IconButton'
import styles from './ModalDialog.module.css'
import { useTranslations } from 'next-intl'

const inter = Inter({ subsets: ['latin'] })

interface ModalDialogProps {
  children?: ReactNode;
  isOpen: boolean;
  'aria-label': string;
  onClose: () => void;
}

const ModalDialog: FC<ModalDialogProps> = ({ 
  children,
  isOpen,
  'aria-label': ariaLabel,
  onClose,
}) => {
  const t = useTranslations()
  const ref = useRef<HTMLDialogElement>(null)

  useEffect(() => {
    const dialog = ref.current
    if (isOpen) {
      console.log(dialog)
      dialog?.showModal()
    } else {
      dialog?.close()
    }
    return () => dialog?.close()
  }, [isOpen]);

  return (
    <dialog 
      className={styles.dialog} 
      ref={ref} 
      aria-label={ariaLabel}
    >
      <IconButton 
        className={styles.closeButton} 
        ariaLabel={t('close')} 
        onClick={onClose} 
      >
        <Icon icon='cancel' />
      </IconButton>
      <div className={`${inter.className} ${styles.content}`}>
        {children}
      </div>
    </dialog>
  )
}

export default ModalDialog