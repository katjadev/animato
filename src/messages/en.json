{
  "projects": "Projects",
  "logout": "Log out",
  "login": "Log in",
  "home-title": "Bring your designs to life with animated SVGs",
  "home-description": "Animato empowers you to easily create and animate scalable vector graphics, breathing life into your designs and enhancing your web presence with eye-catching visuals that capture your audience's attention.",
  "get-started": "Get Started",
  "explore-demo-project": "Explore a demo project",
  "signup-dialog": {
    "create-account": "Create an account",
    "email": "Email",
    "password": "Password"
  },
  "login-dialog": {
    "log-in": "Log in",
    "email": "Email",
    "password": "Password"
  },
  "project-list": {
    "projects": "Projects",
    "profile": "Profile",
    "new-project": "New project",
    "empty-message": "You don't have any projects yet."
  }
}