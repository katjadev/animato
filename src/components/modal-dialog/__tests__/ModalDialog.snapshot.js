import { render } from '@testing-library/react'
import { NextIntlProvider } from 'next-intl'
import messages from '../../../messages/en.json'
import ModalDialog from '../ModalDialog'

jest.mock('next/font/google', () => ({
  Inter: jest.fn().mockImplementation(() => ({
    className: 'inter-class',
  })),
}))

describe('ModalDialog', () => {
  it('renders correctly', () => {
    HTMLDialogElement.prototype.show = jest.fn();
    HTMLDialogElement.prototype.showModal = jest.fn();
    HTMLDialogElement.prototype.close = jest.fn();
    
    const useRouter = jest.spyOn(require('next/router'), 'useRouter')
    const locale = 'en'
    useRouter.mockImplementationOnce(() => ({
      query: { locale: locale },
    }))
  
    const { container } = render(
      <NextIntlProvider messages={messages} locale={locale}>
        <ModalDialog isOpen={true} aria-label='Test Modal' onClose={() => {}}>
          <div>Modal content</div>
        </ModalDialog>
      </NextIntlProvider>
    )
    expect(container).toMatchSnapshot()
  })
})
