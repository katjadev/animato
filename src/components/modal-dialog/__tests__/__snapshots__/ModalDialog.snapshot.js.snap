// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDialog renders correctly 1`] = `
<div>
  <dialog
    aria-label="Test Modal"
    class="dialog"
  >
    <button
      aria-label="Close"
      class="iconButton closeButton"
    >
      <i
        class="iconoir-cancel "
      />
    </button>
    <div
      class="inter-class content"
    >
      <div>
        Modal content
      </div>
    </div>
  </dialog>
</div>
`;
